Branch-SDK/src/io/branch/referral/ServerRequestPing.java

```diff
@@ -0,0 +1,51 @@
+package io.branch.referral;
+
+import android.app.Application;
+import android.content.Context;
+import android.util.Log;
+
+/**
+ * * <p>
+ * The server request for pinging the API to process the queue.
+ * </p>
+ */
+public class ServerRequestPing extends ServerRequest {
+
+    /**
+     * <p>Configures a server request to process the request queue.</p>
+     *
+     * @param context Current {@link Application} context
+     */
+    public ServerRequestPing(Context context) {
+        super(context, null);
+    }
+
+    @Override
+    public boolean handleErrors(Context context) {
+        if (!super.doesAppHasInternetPermission(context)) {
+            Log.i(TAG, "Trouble executing your request. Please add 'android.permission.INTERNET' in your applications manifest file");
```

Comment:
@aaaronlopez can we use `PreferenceHelper#Debug()` instead of log. It take care fo the debug mode and logging level